src/main/kotlin/de/bigboot/ggtools/fang/service/MistforgeService.kt

```diff
@@ -118,33 +156,70 @@ class MistforgeService: AutostartService, KoinComponent {
                 img.overlay(arrow, x, y)
             }
         }
-
+        
         event.message.channel.awaitSingle().createMessageCompat {
             messageReference(event.message.id)
-
-            addEmbedCompat {
-                title(guide.title)
-                thumbnail("https://mistforge.net/img/hero_builder/${guide.heroId}/hero.png")
-                author(guide.username,  "", "https://mistforge.net/customimg/${guide.userId}.png")
-                url("https://mistforge.net/build_viewer?build_id=${buildId}")
-                description(CopyDown().convert(guide.guide).take(150).plus("..."))
-                image("attachment://build.png")
+            if (guide != null) {
+                addEmbedCompat {
+                    title(guide.title)
+                    thumbnail("https://mistforge.net/img/hero_builder/${heroId}/hero.png")
+                    author(guide.username,  "", "https://mistforge.net/customimg/${guide.userId}.png")
+                    url("https://mistforge.net/build_viewer?${params}")
+                    description(CopyDown().convert(guide.guide).take(150).plus("..."))
+                    image("attachment://build.png")
+                }
+            }
+            else {
+                addEmbedCompat {
+                    thumbnail("https://mistforge.net/img/hero_builder/${heroId}/hero.png")
+                    image("attachment://build.png")
+                }
             }
 
-            addComponent(ActionRow.of(ButtonBuildDetails(buildId).component()))
+            addComponent(ActionRow.of(ButtonBuildDetails(params).component()))
 
             addFile("build.png", img.bytes(PngWriter.MinCompression).inputStream())
         }.await()
     }
     private suspend fun handleInteraction(event: ComponentInteractionEvent, button: ButtonBuildDetails) {
         event.deferReply().withEphemeral(true).awaitSafe()
 
-        val guide = mistforge.getGuide(button.buildId.toInt())
-        val hero = heroes.first { it.heroId == guide.heroId }
+        var guide: Guide? = null
+        val query_pairs = HashMap<String, String>()
+        if (button.params.contains("&")) {
+            val pairs = button.params.split("&")
+            for (pair in pairs) {
+                if (pair.contains("=")) {
+                    val idx = pair.indexOf("=")
+                    query_pairs.put(pair.substring(0, idx), pair.substring(idx + 1))
+                }
+            }
+        }
+        if (query_pairs.containsKey("build_id")) {
+            guide = mistforge.getGuide(query_pairs.get("build_id")!!.toInt())
+        }
+
+        var heroId = 0
```

Comment:
use `when` instead of this if/else thing, also rather return without parsing the url than setting some nonsense default